### DISCUSSION TITLE
[✨] API similar to Astro's Content Collections API

### DISCUSSION THREAD

**Discussion by rshackleton** [QwikDev/qwik-evolution#152] [Proposals For Qwik]
### Is your feature request related to a problem?

When working with MDX files it is non-trivial to create dynamic list pages or sitemaps using that data.

### Describe the solution you'd like

An ideal solution would be an equivalent API to Astro's own [Content Collections API](https://docs.astro.build/en/guides/content-collections/).

### Describe alternatives you've considered

I have not yet looked into this but I will be aiming to create a custom vite plugin to read MDX files at compile time and expose the slugs and frontmatter via [a virtual module](https://vitejs.dev/guide/api-plugin.html#virtual-modules-convention).

**Comment by rshackleton** [comment]
I've started creating a package for this here:

https://github.com/rshackleton/rshackleton-qwik/tree/main/packages/vite-mdx-collections

It doesn't include any type generation or validation currently, but it does read and expose the data for use.

**Comment by appinteractive** [comment]
any updates on that issue?

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
